### PR TITLE
[Hunt Tuning] Add Descriptions, Collapse Queries and Re-Generate Docs

### PR DESCRIPTION
**Pull request by terrancedejesus**
## Issues
* https://github.com/elastic/ia-trade-team/issues/280

## Summary
This pull request does the following to the hunting library.

- Adds a description to the hunting schema
- Adds descriptions to all existing hunting query files and docs
- Converts `query` to a `List[str]` for instances where multiple queries may be necessary for the same hunt because of data sources
- Cleans up notes, grammar, language, file names, etc.
- Re-generates all docs

### CODE REVIEW

File: hunting/macos/queries/suspicious_network_connections_by_unsigned_macho.toml [hunk 1]
```diff
@@ -24,4 +28,5 @@ from logs-endpoint.events.network-*
 | keep process.entity_id, process.name, duration_hours, destination.address, MB_out, count_connections, number_of_con_per_hour
 /* threshold is set to 120 connections per minute, you can adjust it to your env/FP rate */
 | where duration_hours >= 8 and number_of_con_per_hour >= 120
-'''
\ No newline at end of file
+'''
+]
```
Comment (Mikaayenson) [inline hunting/macos/queries/suspicious_network_connections_by_unsigned_macho.toml:32]:
```suggestion
]

```

File: hunting/windows/docs/detect_dll_hijack_via_masquerading_as_microsoft_native_libraries.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+# DLL Hijack via Masquerading as Microsoft Native Libraries
+
+---
+
+## Metadata
+
+- **Author:** Elastic
+- **Description:** This hunt identifies when a process loads a DLL normally located in `System32` or `SysWOW64` folders from an unusual path. Adversaries may execute their own malicious payloads by side-loading malicious DLLs. The hosts count also should help exclude false-positives by looking at low occurrences when this abnormal behavior is limited to unique agents.
+- **UUID:** `87c97865-fdaa-48b2-bfa6-67bed7cf56ef`
+- **Integration:** [endpoint](https://docs.elastic.co/integrations/endpoint)
```
Comment (Mikaayenson) [inline hunting/windows/docs/detect_dll_hijack_via_masquerading_as_microsoft_native_libraries.md:10]: Need to add the other integration: I wonder if we should add a check for this.

File: hunting/windows/docs/detect_dll_hijack_via_masquerading_as_microsoft_native_libraries.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+```sql
+from logs-windows.sysmon_operational-*
+| where @timestamp > NOW() - 7 day
+| where host.os.family == "windows" and  event.category == "process" and event.action == "Image loaded" and file.code_signature.status != "Valid" and
+ not file.path rlike """[c-fC-F]:\\(Windows|windows|WINDOWS)\\(System32|SysWOW64|system32|syswow64)\\[a-zA-Z0-9_]+.dll"""
+| keep file.name, file.path, file.hash.sha256, process.executable, host.id
+ /* steps to create DL enrichment policy https://gist.github.com/Samirbous/9f9c3237a0ada745e71cc2ba3425311c - just replace dll by file */
+| ENRICH libs-policy-sysmon
```
Comment (Mikaayenson) [inline hunting/windows/docs/detect_dll_hijack_via_masquerading_as_microsoft_native_libraries.md:39]: unrelated to this pr, I wonder if we should do in line lookup

File: hunting/windows/docs/libraries_loaded_by_svchost_with_low_occurrence_frequency.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+---
+
+## Metadata
+
+- **Author:** Elastic
+- **Description:** This hunt returns the SHA256 hash and the `dll.path` of unsigned libraries loaded by svchost where the presence of unique path/hash is limited to a unique host. Adversaries may use Windows service DLLs to maintain persistence or run with System privileges.
+
+- **UUID:** `e37fe0b9-1b70-4800-8989-58bac5a0a9bb`
+- **Integration:** [endpoint](https://docs.elastic.co/integrations/endpoint)
```
Comment (Mikaayenson) [inline hunting/windows/docs/libraries_loaded_by_svchost_with_low_occurrence_frequency.md:11]: Windows integration here?

File: hunting/windows/queries/drivers_load_with_low_occurrence_frequency.toml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+language = "ES|QL"
+license = "Elastic License v2"
+notes = [
+    "This hunt has three optional queries, one for Elastic Defend data, another for Sysmon data and the last one for Windows 7045 events.",
+    "Further investigation can be done pivoting by `dll.pe.imphash` or `dll.name.`",
+    "`dll.Ext.relative_file_creation_time` is used in the first query to limit the result to recently dropped drivers (populated in Elastic Defend).",
+    "Aggregation can also be done by `dll.hash.sha256` / `file.hash.sha256` but will return more results.",
+    "Bring Your Own Vulnerable Driver (BYOVD) are all signed and not malicious, further investigation should be done to check the surrounding events (service creation, process that dropped the driver etc.).",
+]
+mitre = [ "T1068",]
```
Comment (Mikaayenson) [inline hunting/windows/queries/drivers_load_with_low_occurrence_frequency.toml:18]:
```suggestion
mitre = [ "T1068"]
```

File: hunting/windows/queries/domain_names_queried_via_lolbins_and_with_low_occurence_frequency.toml [hunk 6]
```diff
@@ -1,22 +1,26 @@
...
+name = "DNS Queries via LOLBins with Low Occurence Frequency"
 language = "ES|QL"
 license = "Elastic License v2"
 notes = [
-    "Utilities like curl and SSL verification web-servvices are noisy, while others are rare like scripting utilities and are worth further investigation.",
-    "Connection to legit domains like github, discord, telegram and many other legit web-services by lolbins is still suspicious and require further investigation.",
+    "Utilities like curl and SSL verification for web services are noisy, while others are rare such as scripting utilities and are worth further investigation.",
+    "Connection to legit domains like Github, Discord, Telegram and many other legit web services by LOLBins is still suspicious and require further investigation.",
 ]
 mitre = [ "T1071",]
```
Comment (Mikaayenson) [inline hunting/windows/queries/domain_names_queried_via_lolbins_and_with_low_occurence_frequency.toml:15]:
```suggestion
mitre = [ "T1071"]
```

File: hunting/windows/queries/excessive_rdp_network_activity_by_source_host_and_user.toml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+uuid = "fe01a8a5-6367-4c4c-a57b-be513ab80e42"
+name = "Excessive RDP Network Activity by Host and User"
+language = "ES|QL"
+license = "Elastic License v2"
+notes = [
+    "Further investigation can done pivoting by `host.id` and `user.name`.",
+    "Depending on normal SysAdmin RDP activity, the threshold of 10 can be adjusted to reduce normal noisy activity.",
+    "The second query uses Windows Security log event ID 4624 to summarize numbers of RDP connections by `source.ip` and `user.name` and duration."
+]
+mitre = [ "T1021", "T1021.001",]
```
Comment (Mikaayenson) [inline hunting/windows/queries/excessive_rdp_network_activity_by_source_host_and_user.toml:16]:
```suggestion
mitre = [ "T1021", "T1021.001"]
```

Not sure if these need to be all double checked.

File: hunting/windows/queries/excessive_rdp_network_activity_by_source_host_and_user.toml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+''',
+'''
+from logs-endpoint.events.network-*, logs-windows.sysmon_operational-*
+| where  @timestamp > now() - 7 day
+| where host.os.family == "windows" and event.category == "network" and process.name == "svchost.exe" and network.direction == "ingress" and
+  network.transport == "tcp"and destination.port == 3389 and source.port >= 49152
+| stats agents = count_distinct(host.id) by source.ip
+| where agents >= 10
+'''
+]
```
Comment (Mikaayenson) [inline hunting/windows/queries/excessive_rdp_network_activity_by_source_host_and_user.toml:37]:
```suggestion
]

```

**Review by Mikaayenson — APPROVED** [review]
Great going through each of these.

I added some nits but not in every example, in general (extra comma on the mitre tags, and the newline at the end of file. 

There were also two examples of integrations that should be added properly. 

Otherwise lgtm.

File: hunting/windows/docs/detect_dll_hijack_via_masquerading_as_microsoft_native_libraries.md (hunk 2 above)
Comment (terrancedejesus) [reply hunting/windows/docs/detect_dll_hijack_via_masquerading_as_microsoft_native_libraries.md:10]: Fixed.

File: hunting/windows/docs/libraries_loaded_by_svchost_with_low_occurrence_frequency.md (hunk 4 above)
Comment (terrancedejesus) [reply hunting/windows/docs/libraries_loaded_by_svchost_with_low_occurrence_frequency.md:11]: Fixed.

**Comment by terrancedejesus** [comment]
@brokensound77  thanks for commenting on this PR.

> you lose the index-to-query mapping, where you now have 2 arrays of data with ambiguous relationships
I do not recall the goal being to have a "index-to-query" mapping relationship. Rather a single hunt could contain multiple queries. We also do not include indexes in the hunting schema, as the expected query language is ES|QL where the index patterns are included in the `from` command. If we want to expand these hunts to use other query languages, we can add `datasources` or `index` to the hunt schema and have it be optional if KQL or EQL are used. However, that should be a future improvement we can add to the roadmap and should not block this PR.

> you lose the unique UUID-to-query mapping
Yes, there is no UUID-to-query mapping, but I do not recall this being a requirement either. The UUIDs should map to their related hunts, not queries.

As we originally discussed, we are attempting to avoid over-engineering this as it's sole purpose is to share hunting analytics publicly and reach MVP. Overtime, we can look to add new features, updated schemas, etc. as needed since this is not a product-driven initiative.

File: hunting/windows/queries/executable_file_creation_by_an_unusual_microsoft_binary.toml [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+integration = ["endpoint", "windows"]
+uuid = "3b2900fe-74d9-4c49-b3df-cbeceb02e841"
+name = "Executable File Creation by an Unusual Microsoft Binary"
+language = "ES|QL"
+license = "Elastic License v2"
+notes = [
+    "This hunt includes two optional queries, one for Elastic Defend data and another for Sysmon data.",
+    "Sysmon file events don't populate file header and process code signature information thus we use `file.extension`.",
+    "Some exploits may result in the creation of an executable file by the exploited process.",
+    "Further investigation can be done by pivoting on`process.executable` and filtering for executable file creation.",
```
Comment (w0rk3r) [inline hunting/windows/queries/executable_file_creation_by_an_unusual_microsoft_binary.toml:16]:
```suggestion
    "Further investigation can be done by pivoting on `process.executable` and filtering for executable file creation.",
```

File: hunting/windows/queries/execution_via_windows_management_instrumentation_by_occurrence_frequency_by_unique_agent.toml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+license = "Elastic License v2"
+notes = [
+    "This hunt contains three queries for Elastic Defend, Sysmon, and Windows Security event 4688."]
+mitre = [ "T1047"]
+query = [
+'''
+from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*
+| where @timestamp > now() - 7 day and
+  host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation") and
+  process.parent.name.caseless == "wmiprvse.exe" and starts_with(process.code_signature.subject_name, "Microsoft")
```
Comment (w0rk3r) [inline hunting/windows/queries/execution_via_windows_management_instrumentation_by_occurrence_frequency_by_unique_agent.toml:19]: I don't think these would work as only Elastic Defend has caseless fields atm

File: hunting/windows/queries/microsoft_office_child_processes_with_low_occurrence_frequency.toml [hunk 11]
```diff
@@ -1,23 +1,27 @@
...
-query = '''
+mitre = [ "T1566", "T1566.001"]
+query = [
+'''
 from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*, logs-system.security-*
-| where host.os.family == "windows" and @timestamp > NOW() - 15 day and 
-  event.category == "process" and event.action in ("start", "Process creation", "created-process") and 
+| where host.os.family == "windows" and @timestamp > NOW() - 15 day and
+  event.category == "process" and event.action in ("start", "Process creation", "created-process") and
   process.parent.name.caseless in ("winword.exe", "excel.exe", "powerpnt.exe") and not starts_with(process.executable, "C:\\Program Files")
```
Comment (w0rk3r) [inline hunting/windows/queries/microsoft_office_child_processes_with_low_occurrence_frequency.toml:21]: I don't think these would work as only Elastic Defend has caseless fields atm

File: hunting/windows/queries/rundll32_execution_aggregated_by_cmdline.toml [hunk 12]
```diff
@@ -1,25 +1,29 @@
...
-query = '''
+mitre = [ "T1127", "T1218", "T1218.011"]
+query = [
+'''
 from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*, logs-system.security-*
 | where  @timestamp > now() - 7 day
-| where host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation", "created-process") and 
-  process.name.caseless == "rundll32.exe" and 
+| where host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation", "created-process") and
+  process.name.caseless == "rundll32.exe" and
```
Comment (w0rk3r) [inline hunting/windows/queries/rundll32_execution_aggregated_by_cmdline.toml:21]: I don't think these would work as only Elastic Defend has caseless fields atm

File: hunting/windows/queries/windows_command_and_scripting_interpreter_from_unusual_parent.toml [hunk 13]
```diff
@@ -1,23 +1,27 @@
...
-query = '''
+mitre = [ "T1059", "T1059.001", "T1059.003"]
+query = [
+'''
 from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*, logs-system.security-*
 | where  @timestamp > now() - 7 day
-| where host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation", "created-process") and 
-  process.name.caseless in ("cmd.exe", "powershell.exe", "conhost.exe") and 
+| where host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation", "created-process") and
+  process.name.caseless in ("cmd.exe", "powershell.exe", "conhost.exe") and
```
Comment (w0rk3r) [inline hunting/windows/queries/windows_command_and_scripting_interpreter_from_unusual_parent.toml:21]: I don't think these would work as only Elastic Defend has caseless fields atm

File: hunting/windows/queries/detect_dll_hijack_via_masquerading_as_microsoft_native_libraries.toml [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+description = "This hunt identifies when a process loads a DLL normally located in `System32` or `SysWOW64` folders from an unusual path. Adversaries may execute their own malicious payloads by side-loading malicious DLLs. The hosts count also should help exclude false-positives by looking at low occurrences when this abnormal behavior is limited to unique agents."
+integration = ["endpoint", "windows"]
+uuid = "87c97865-fdaa-48b2-bfa6-67bed7cf56ef"
+name = "DLL Hijack via Masquerading as Microsoft Native Libraries"
+language = "ES|QL"
+license = "Elastic License v2"
+mitre = ["T1574", "T1574.001"]
+notes = [
+    "This hunt has two optional queries, one for Elastic Defend data and another for Sysmon data.",
+    "This hunt require the creation of an enrichment policy to use with the ES|QL (ENRICH command).",
```
[the remark this answers is not in the document]
Comment (terrancedejesus) [reply hunting/windows/queries/detect_dll_hijack_via_masquerading_as_microsoft_native_libraries.toml:12]: That is a great point, I'll link it and any others that reference the enrichment policy.

**Comment by brokensound77** [comment]
> @brokensound77 thanks for commenting on this PR.
> 
> > you lose the index-to-query mapping, where you now have 2 arrays of data with ambiguous relationships
> 
> I do not recall the goal being to have a "index-to-query" mapping relationship. Rather a single hunt could contain multiple queries. We also do not include indexes in the hunting schema, as the expected query language is ES|QL where the index patterns are included in the `from` command. If we want to expand these hunts to use other query languages, we can add `datasources` or `index` to the hunt schema and have it be optional if KQL or EQL are used. However, that should be a future improvement we can add to the roadmap and should not block this PR.

We can certainly punt it until we have non-ES|QL, but it is inevitable, where at some point we must address it. If we leave it ambiguous, then it almost defeats the purpose of `integration` as a dedicated field at all (since, as you stated, its already in the query). This PR _introduces_ the ambiguity, however, so that is worth considering what to punt vs address here.

Can you share how you see `integration` being used - perhaps misalignment in its use is leading to our discrepancies?

The idea is the allow this to be as programmatically compatible as possible (both for automation outside of the stack, as well as eventual features within the stack). 



> > you lose the unique UUID-to-query mapping
> 
> Yes, there is no UUID-to-query mapping, but I do not recall this being a requirement either. The UUIDs should map to their related hunts, not queries.
> 
> As we originally discussed, we are attempting to avoid over-engineering this as it's sole purpose is to share hunting analytics publicly and reach MVP. Overtime, we can look to add new features, updated schemas, etc. as needed since this is not a product-driven initiative.


The idea here is that each query is uniquely identifiable for programmatic use and testing. This coupled with ambiguous required integrations seems to handicap automating simply on top of this (especially since we do not have support for ES|QL parsing).

So if someone's goal was to load all queries, and run them only on compatible/available datasources, and identify any that have results or issues, avoidable problems would hinder this.


I don't think keeping a 1-1 UID falls into over-engineering. Thoughts?


---

Short of getting our backlog of queries pushed, is there something else driving the completion date of these? Especially to justify punting some design decisions?

**Comment by terrancedejesus** [comment]
@brokensound77 - We agreed that we can turn some of these thoughtful discussions into issues where we can add enhancement to the hunts at a later time.

File: hunting/windows/queries/windows_command_and_scripting_interpreter_from_unusual_parent.toml (hunk 13 above)
Comment (Samirbous) [inline hunting/windows/queries/windows_command_and_scripting_interpreter_from_unusual_parent.toml:21]:
```suggestion
  to_lower(process.name) in ("cmd.exe", "powershell.exe", "conhost.exe") and
```

File: hunting/windows/queries/windows_command_and_scripting_interpreter_from_unusual_parent.toml [hunk 15]
```diff
@@ -1,23 +1,27 @@
...
   (starts_with(process.parent.executable.caseless, "c:\\windows\\system32") or starts_with(process.parent.executable.caseless, "c:\\windows\\syswow64"))
```
Comment (Samirbous) [inline hunting/windows/queries/windows_command_and_scripting_interpreter_from_unusual_parent.toml:22]:
```suggestion
  (starts_with(to_lower(process.parent.executable), "c:\\windows\\system32") or starts_with(to_lower(process.parent.executable), "c:\\windows\\syswow64"))
```

File: hunting/windows/docs/detect_rare_lsass_process_access_attempts.md [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+- **Language:** `ES|QL`
+
+## Query
+
+```sql
+from logs-endpoint.events.api*
+| where  @timestamp > NOW() - 7 day
+| where event.category == "api" and host.os.family == "windows" and process.Ext.api.name in ("OpenProcess", "OpenThread", "ReadProcessMemory") and
+ Target.process.name == "lsass.exe"
+| keep process.executable.caseless, host.id
```
Comment (Samirbous) [inline hunting/windows/docs/detect_rare_lsass_process_access_attempts.md:21]:
```suggestion
| keep process.executable, host.id
```

File: hunting/windows/docs/detect_rare_lsass_process_access_attempts.md [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+ /* normalize process paths to reduce known random patterns in process.executable */
+| eval process = replace(process.executable.caseless, """([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}|ns[a-z][A-Z0-9]{3,4}\.tmp|DX[A-Z0-9]{3,4}\.tmp|7z[A-Z0-9]{3,5}\.tmp|[0-9\.\-\_]{3,})""", "")
```
Comment (Samirbous) [inline hunting/windows/docs/detect_rare_lsass_process_access_attempts.md:23]:
```suggestion
| eval process = replace(to_lower(process.executable), """([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}|ns[a-z][A-Z0-9]{3,4}\.tmp|DX[A-Z0-9]{3,4}\.tmp|7z[A-Z0-9]{3,5}\.tmp|[0-9\.\-\_]{3,})""", "")
```

File: hunting/windows/docs/execution_via_remote_services_by_client_address.md [hunk 18]
```diff
@@ -13,10 +15,10 @@
...
 ```sql
 from logs-endpoint.events.process-*
-| where  @timestamp > now() - 7 day and host.os.family == "windows" and 
-  event.category == "process" and event.action == "start" and 
+| where  @timestamp > now() - 7 day and host.os.family == "windows" and
+  event.category == "process" and event.action == "start" and
   /* network logon type */
-  process.Ext.session_info.logon_type == "Network" and 
+  process.Ext.session_info.logon_type == "Network" and
   (process.parent.name .caseless in ("wmiprvse.exe", "wsmprovhost.exe", "winrshost.exe") or (process.parent.name == "svchost.exe" and process.parent.args == "DcomLaunch"))
```
Comment (Samirbous) [inline hunting/windows/docs/execution_via_remote_services_by_client_address.md:22]:
```suggestion
  (to_lower(process.parent.name) in ("wmiprvse.exe", "wsmprovhost.exe", "winrshost.exe") or (process.parent.name == "svchost.exe" and process.parent.args == "DcomLaunch"))
```

File: hunting/windows/docs/execution_via_windows_management_instrumentation_by_occurrence_frequency_by_unique_agent.md [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+| stats agents = count_distinct(host.id) by process.name
+| where agents == 1
+```
+
+```sql
+from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*, logs-system.security-*
+| where  @timestamp > now() - 7 day and
+  host.os.family == "windows" and event.category == "process" and
+  event.action in ("start", "Process creation", "created-process") and
+  process.parent.name.caseless == "wmiprvse.exe"
```
Comment (Samirbous) [inline hunting/windows/docs/execution_via_windows_management_instrumentation_by_occurrence_frequency_by_unique_agent.md:31]:
```suggestion
  to_lower(process.parent.name) == "wmiprvse.exe"
```

File: hunting/windows/docs/execution_via_windows_management_instrumentation_by_occurrence_frequency_by_unique_agent.md [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+| eval cmdline = replace(cmdline, """([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}|ns[a-z][A-Z0-9]{3,4}\.tmp|DX[A-Z0-9]{3,4}\.tmp|7z[A-Z0-9]{3,5}\.tmp|[0-9\.\-\_]{3,})""", "")
+| stats agents = count_distinct(host.id) by cmdline
+| where agents == 1
+```
+
+```sql
+from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*
+| where @timestamp > now() - 7 day and
+  host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation") and
+  process.parent.name.caseless == "wmiprvse.exe" and (process.code_signature.exists == false or process.code_signature.trusted == false)
```
Comment (Samirbous) [inline hunting/windows/docs/execution_via_windows_management_instrumentation_by_occurrence_frequency_by_unique_agent.md:43]:
```suggestion
  to_lower(process.parent.name) == "wmiprvse.exe" and (process.code_signature.exists == false or process.code_signature.trusted == false)
```

File: hunting/windows/docs/microsoft_office_child_processes_with_low_occurrence_frequency.md [hunk 21]
```diff
@@ -13,18 +15,18 @@
 
 ```sql
 from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*, logs-system.security-*
-| where host.os.family == "windows" and @timestamp > NOW() - 15 day and 
-  event.category == "process" and event.action in ("start", "Process creation", "created-process") and 
+| where host.os.family == "windows" and @timestamp > NOW() - 15 day and
+  event.category == "process" and event.action in ("start", "Process creation", "created-process") and
   process.parent.name.caseless in ("winword.exe", "excel.exe", "powerpnt.exe") and not starts_with(process.executable, "C:\\Program Files")
 // normalize user home profile paths
 | eval process_path = replace(process.executable.caseless, """[c]:\\[u][s][e][r][s]\\[a-zA-Z0-9\.\-\_\$]+\\""", "c:\\\\users\\\\user\\\\")
```
Comment (Samirbous) [inline hunting/windows/docs/microsoft_office_child_processes_with_low_occurrence_frequency.md:22]:
```suggestion
| eval process_path = replace(to_lower(process.executable), """[c]:\\[u][s][e][r][s]\\[a-zA-Z0-9\.\-\_\$]+\\""", "c:\\\\users\\\\user\\\\")
```

File: hunting/windows/docs/microsoft_office_child_processes_with_low_occurrence_frequency.md [hunk 22]
```diff
@@ -13,18 +15,18 @@
 
 ```sql
 from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*, logs-system.security-*
-| where host.os.family == "windows" and @timestamp > NOW() - 15 day and 
-  event.category == "process" and event.action in ("start", "Process creation", "created-process") and 
+| where host.os.family == "windows" and @timestamp > NOW() - 15 day and
+  event.category == "process" and event.action in ("start", "Process creation", "created-process") and
   process.parent.name.caseless in ("winword.exe", "excel.exe", "powerpnt.exe") and not starts_with(process.executable, "C:\\Program Files")
```
Comment (Samirbous) [inline hunting/windows/docs/microsoft_office_child_processes_with_low_occurrence_frequency.md:20]:
```suggestion
  to_lower(process.parent.name) in ("winword.exe", "excel.exe", "powerpnt.exe") and not starts_with(process.executable, "C:\\Program Files")
```

File: hunting/windows/docs/rundll32_execution_aggregated_by_cmdline.md [hunk 23]
```diff
@@ -14,8 +16,8 @@
 ```sql
 from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*, logs-system.security-*
 | where  @timestamp > now() - 7 day
-| where host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation", "created-process") and 
-  process.name.caseless == "rundll32.exe" and 
+| where host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation", "created-process") and
+  process.name.caseless == "rundll32.exe" and
```
Comment (Samirbous) [inline hunting/windows/docs/rundll32_execution_aggregated_by_cmdline.md:20]:
```suggestion
  to_lower(process.name) == "rundll32.exe" and
```

File: hunting/windows/docs/windows_command_and_scripting_interpreter_from_unusual_parent.md [hunk 24]
```diff
@@ -14,8 +16,8 @@
 ```sql
 from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*, logs-system.security-*
 | where  @timestamp > now() - 7 day
-| where host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation", "created-process") and 
-  process.name.caseless in ("cmd.exe", "powershell.exe", "conhost.exe") and 
+| where host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation", "created-process") and
+  process.name.caseless in ("cmd.exe", "powershell.exe", "conhost.exe") and
```
Comment (Samirbous) [inline hunting/windows/docs/windows_command_and_scripting_interpreter_from_unusual_parent.md:20]:
```suggestion
  to_lower(process.name) in ("cmd.exe", "powershell.exe", "conhost.exe") and
```

File: hunting/windows/docs/windows_command_and_scripting_interpreter_from_unusual_parent.md [hunk 25]
```diff
@@ -14,8 +16,8 @@
...
   (starts_with(process.parent.executable.caseless, "c:\\windows\\system32") or starts_with(process.parent.executable.caseless, "c:\\windows\\syswow64"))
```
Comment (Samirbous) [inline hunting/windows/docs/windows_command_and_scripting_interpreter_from_unusual_parent.md:21]:
```suggestion
  (starts_with(to_lower(process.parent.executable), "c:\\windows\\system32") or starts_with(to_lower(process.parent.executable), "c:\\windows\\syswow64"))
```

File: hunting/windows/queries/execution_via_windows_management_instrumentation_by_occurrence_frequency_by_unique_agent.toml (hunk 10 above)
Comment (Samirbous) [reply hunting/windows/queries/execution_via_windows_management_instrumentation_by_occurrence_frequency_by_unique_agent.toml:19]: using `to_lower` which was added in 8.14

File: hunting/windows/queries/microsoft_office_child_processes_with_low_occurrence_frequency.toml (hunk 11 above)
Comment (Samirbous) [reply hunting/windows/queries/microsoft_office_child_processes_with_low_occurrence_frequency.toml:21]: using `to_lower` which was added in 8.14

File: hunting/windows/docs/execution_via_windows_management_instrumentation_by_occurrence_frequency_by_unique_agent.md [hunk 26]
```diff
@@ -0,0 +1,58 @@
...
+- **Integration:** [endpoint](https://docs.elastic.co/integrations/endpoint), [windows](https://docs.elastic.co/integrations/windows), [system](https://docs.elastic.co/integrations/system)
+- **Language:** `ES|QL`
+
+## Query
+
+```sql
+from logs-endpoint.events.process-*, logs-windows.sysmon_operational-*
+| where @timestamp > now() - 7 day and
+  host.os.family == "windows" and event.category == "process" and event.action in ("start", "Process creation") and
+  process.parent.name.caseless == "wmiprvse.exe" and starts_with(process.code_signature.subject_name, "Microsoft")
```
Comment (Samirbous) [inline hunting/windows/docs/execution_via_windows_management_instrumentation_by_occurrence_frequency_by_unique_agent.md:20]:
```suggestion
  to_lower(process.parent.name) == "wmiprvse.exe" and starts_with(process.code_signature.subject_name, "Microsoft")
```

**Review by Samirbous — APPROVED** [review]

**Pull request merged**